controllers/layered_image_handler.go

```diff
@@ -0,0 +1,149 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	ignTypes "github.com/coreos/ignition/v2/config/v3_2/types"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	LayeredImageDeployCm = "layered-image-deploy-cm"
+)
+
+// Global struct to store the Image based MachineConfig struct
+var ImgMc *mcfgv1.MachineConfig
+
+func (r *KataConfigOpenShiftReconciler) handleLayeredImageDeploymentFeature(state FeatureGateState) error {
+
+	// Check if MachineConfig exists and return the same without changing anything
+	mc, err := r.getExistingMachineConfig()
```

Comment:
Is this needed? The possibility of a MachineConfig already existing is managed in the function `createMc`.

Reply:
Let's take a step back and see if we can have a consensus on the basic pattern that needs to be used for setting the FG configs. 

1. Pattern-1: Check if FG is enabled at the reconciliation loop's beginning and set FG-specific variables (e.g., using ImgMc as part of the KataConfigOpenShiftReconciler struct or otherwise ). If needed, use the required FG-specific variables later in the reconciliation loop.
2. Pattern-2: Where ever needed in the reconcile loop, check if FG is enabled and set and use the required FG-specific variables

I find pattern-1 easier to understand and maintain going forward for different FGs. 

@spotlesstofu @pmores @snir911 do you see any downsides in using pattern-1 ? Do you have a strong preference for pattern-2 ?

Reply:
Nothing obvious jump in my mind, i'm find with pattern-1

Reply:
@pmores @spotlesstofu can I proceed with pattern-1 and complete the pending changes?

Reply:
Yes. Sorry for not answering before.